pkg/common/redis_test.go

```diff
@@ -87,12 +87,133 @@ func TestRedisLockWithTTLAndRetry(t *testing.T) {
 }
 
 func TestCopyStruct(t *testing.T) {
-	options1 := &types.RedisConfig{ClientName: "hello", PoolSize: 10, ConnMaxLifetime: time.Second}
-	options2 := &types.RedisConfig{}
+	tests := []struct {
+		name     string
+		src      any
+		dest     any
+		expected any
+	}{
+		{
+			name: "copy struct basic",
+			src: &types.RedisConfig{
+				ClientName:      "hello",
+				PoolSize:        10,
+				ConnMaxLifetime: time.Second,
+			},
+			dest: &types.RedisConfig{},
+			expected: &types.RedisConfig{
+				ClientName:      "hello",
+				PoolSize:        10,
+				ConnMaxLifetime: time.Second,
+			},
+		},
+		{
+			name: "copy struct overwriting fields on destination",
+			src: &types.RedisConfig{
+				ClientName:      "hello",
+				PoolSize:        10,
+				ConnMaxLifetime: time.Second,
+			},
+			dest: &types.RedisConfig{
+				ClientName: "world",
+				PoolSize:   20,
+			},
+			expected: &types.RedisConfig{
+				ClientName:      "hello",
+				PoolSize:        10,
+				ConnMaxLifetime: time.Second,
+			},
+		},
+		{
+			name: "copy struct overwriting fields on destination part 2",
+			src: &types.Worker{
+				Id:       "123",
+				Status:   types.WorkerStatusAvailable,
+				TotalCpu: 10,
+			},
+			dest: &types.Worker{
+				Id:           "456",
+				Status:       types.WorkerStatusPending,
+				TotalCpu:     20,
+				TotalMemory:  0,
+				BuildVersion: "0.1.0",
+			},
+			expected: &types.Worker{
+				Id:                   "123",
+				Status:               types.WorkerStatusAvailable,
+				TotalCpu:             10,
+				TotalMemory:          0,
```

Comment:
nit-pick but i think it would be worth making all of these non-zero values

Reply:
This is a good library if you want to automate creating a bunch of fake structs a bit: https://github.com/go-faker/faker

Reply:
Yeah good call. The test after this tests for defaults. I've updated this.